README.md

```diff
@@ -53,4 +84,175 @@ To generate and access the API documentation on backend, follow these steps:
+- **Album Filtering**: Filter albums based on criteria such as album name and artist name.
+- **Spotify URL Validation**: Ensure the validity of Spotify URLs associated with albums.
+- **Virtual Fields**: Utilize virtual fields to track users who have purchased specific albums.
+- **Automatic Album Updates**: Automatically update album information and manage cancellations.
+- **Tax Management**: Simplify tax calculations and management to ensure accurate album pricing.
+
+### Review Management:
+
+- **Creating, updating, and deleting reviews**: Easily create new reviews, update existing ones, and delete them using CRUD operations for efficient review management.
+- **Fetching all reviews, reviews for a specific product, or a single review**: Access the complete list of all reviews, reviews for a specific product, or a selected review for effective tracking and management.
+- **Preventing users from submitting multiple reviews for the same product**: Ensure that users cannot submit multiple reviews for the same product.
+
+### Wishlist Management:
+
+- **Creating a Wishlist**: Create a user's wishlist, or retrieve an existing one if it already exists.
+- **Adding and Removing Albums**: Add albums to a user's wishlist or remove them as needed.
+- **Fetching a User's Wishlist**: Retrieve a user's wishlist to view the albums they've added.
+
+### Order Management:
+
+- **Creating Album Orders**: Easily generate album orders with detailed pricing, utilizing CRUD operations for efficient management.
+- **Order Retrieval**: Access and retrieve specific order details for effective tracking and management.
+- **Stripe Integration**: Securely process album order payments via Stripe, ensuring a safe and convenient checkout.
+- **Automatic Order Updates and TTL Functionality**: Automatically update orders, cancel unpaid ones after a defined time for accurate statuses, and remove canceled orders after a set duration for streamlined data management.
+- **Tax Management**: Simplify tax calculations and management for precise order pricing.
+
+### Admin Dashboard:
+
+- A simple admin dashboard that allows administrators to update album prices.
+  ![Admin Page]
+  <img src="images/adminPage.png" alt="admin page" style="float: left; margin-right: 10px;" />
+
+## Prerequisites
+
+Before running the Music Store API, ensure you have the following prerequisites installed:
+
+- **Node.js**: You can download and install Node.js from the official website: [Node.js Downloads](https://nodejs.org/en/download/).
+
+- **MongoDB**: You should have a MongoDB server running locally or provide a connection URI to an external MongoDB server. You can download MongoDB from the official website: [MongoDB Downloads](https://www.mongodb.com/try/download/community).
+
+- **Stripe Account**: To enable payment processing, you'll need to sign up for a Stripe account and obtain the API secret key. Visit the Stripe website: [Stripe](https://stripe.com/).
```

Comment:
Also a spotify app credentials, which can be set up using these instructions: https://developer.spotify.com/documentation/web-api